modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManagerTest.java

```diff
@@ -0,0 +1,191 @@
+    public void testListWithGetInt(String filter, int startIndexInt, int countInt,
+                                   String sortBy, String sortOrder, String domainName, String attributes,
+                                   String excludeAttributes, int expectedScimResponseStatus) {
+
+        SCIMResponse outputScimResponse = userResourceManager.listWithGET(userManager, filter, startIndexInt,
+                countInt, sortBy, sortOrder, domainName, attributes, excludeAttributes);
+
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+    }
+
+    @DataProvider(name = "dataForListWithGetInteger")
+    public Object[][] dataToGetListInteger() throws CharonException, BadRequestException, InternalErrorException {
+
+        return new Object[][]{
+                {null, 1, 2, null, null, "PRIMARY", "emails", null, 200},
+                {"userName sw Rash", 1, 2, null, null, "PRIMARY", "userName,name.familyName",
+                        "emails", 200}
+        };
+    }
+
+    @Test(dataProvider = "dataForListWithGetInteger")
+    public void testListWithGetInteger(String filter, Integer startIndexInt,
+                        Integer countInt, String sortBy, String sortOrder, String domainName,
+                        String attributes, String excludeAttributes, int expectedScimResponseStatus) {
+
+        SCIMResponse outputScimResponse = userResourceManager.listWithGET(userManager, filter, startIndexInt,
+                countInt, sortBy, sortOrder, domainName, attributes, excludeAttributes);
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+    }
+
+    private User getNewUser() throws BadRequestException, CharonException, InternalErrorException {
+
+        String scimObjectString = "{\n" +
+                "  \"schemas\": \n" +
+                "    [\"urn:ietf:params:scim:schemas:core:2.0:User\",\n" +
+                "    \"urn:ietf:params:scim:schemas:extension:enterprise:2.0:User\"\n" +
+                "    ],\n" +
+                " \"name\": {\n" +
+                "    \"givenName\": \"Rash\",\n" +
+                "    \"familyName\": \"Na\",\n" +
+                "    \"formatted\": \"Rash Na\"\n" +
+                "  },\n" +
+                "  \"userName\": \"rash\",\n" +
+                "  \"password\": \"rash123\",\n" +
+                "  \"id\": \"123\",\n" +
+                "  \"emails\": [\n" +
+                "      {\n" +
+                "        \"type\": \"home\",\n" +
+                "        \"value\": \"rash@gmail.com\",\n" +
+                "         \"primary\": true\n" +
+                "      }\n" +
+                "  ]\n" +
+                "}";
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+        User user = (User) decoder.decodeResource(scimObjectString, schema, new User());
```

Comment:
There is no need to cast the object. `decodeResource` should already be returning an object of type `User`

Reply:
fixed in 9c8cc7303598fb78fc609acbe4cc7f14965a1737